api/Infrastructure/Authentication/AuthenticationServiceCollectionExtension.cs

```diff
@@ -271,5 +245,92 @@ await cookieCtx.HttpContext.SignOutAsync(CookieAuthenticationDefaults
 
             return services;
         }
+
+        private static async Task OnPostAuthSuccess(
+            IConfiguration configuration,
+            Microsoft.AspNetCore.Authentication.OpenIdConnect.TokenValidatedContext context,
+            ILogger logger,
+            List<Claim> claims)
+        {
+            var judgeId = configuration.GetNonEmptyValue("PCSS:JudgeId");
+            var homeLocationId = configuration.GetNonEmptyValue("PCSS:JudgeHomeLocationId");
+
+            // Remove checking when the "real" mongo db has been configured
+            var connectionString = configuration.GetValue<string>("MONGODB_CONNECTION_STRING");
+            if (string.IsNullOrWhiteSpace(connectionString))
+            {
+                // Defaults the logged in user to the default judge.
+                logger.LogInformation("Acting as Judge Id - {JudgeId}.", judgeId);
+                claims.Add(new Claim(CustomClaimTypes.JudgeId, judgeId));
+                claims.Add(new Claim(CustomClaimTypes.JudgeHomeLocationId, homeLocationId));
+                return;
+            }
+
+            try
+            {
+                var userService = context.HttpContext.RequestServices.GetRequiredService<IUserService>();
+                var userDto = await userService.GetWithPermissionsAsync(context.Principal.Email());
```

Comment:
@ronaldo-macapobre Looking through this again and I'm not sure this is safe, as this will create a new user if the user's email changes. I think in most cases email is stable but may not always be - I know in the case of sso standard they recommend that the preferred_username guid is used instead.